### PR TITLE
Add support for the `background-color` attribute

### PR DESCRIPTION
**Pull request by LaurenzV**

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Is this task finished?

**Comment by LaurenzV** [comment]
I think so? 😄

**Comment by RazrFalcon** [comment]
Good. Just wanted to clarify.

File: crates/resvg/tests/integration/main.rs [hunk 1]
```diff
@@ -55,10 +57,14 @@ pub fn render(name: &str) -> usize {
     );
     resvg::render(&tree, render_ts, &mut pixmap.as_mut());
 
-    // pixmap.save_png(&format!("tests/{}.png", name)).unwrap();
-    // Command::new("oxipng")
-    //     .args(["-o".to_owned(), "6".to_owned(), "-Z".to_owned(), format!("tests/{}.png", name)])
-    //     .output().unwrap();
+    if !Path::new(&png_path).exists() {
```
Comment (RazrFalcon) [inline crates/resvg/tests/integration/main.rs:60]: Let's revert to the old implementation. I want this behavior to be manually trigger-able.

File: crates/usvg/src/parser/converter.rs [hunk 2]
```diff
@@ -7,15 +7,15 @@ use std::hash::{Hash, Hasher};
...
-#[cfg(feature = "text")]
-use fontdb::Database;
-use svgtypes::{Length, LengthUnit as Unit, PaintOrderKind, TransformOrigin};
-
 use super::svgtree::{self, AId, EId, FromValue, SvgNode};
 use super::units::{self, convert_length};
 use super::{marker, Error, Options};
 use crate::parser::paint_server::process_paint;
 use crate::*;
+#[cfg(feature = "text")]
```
Comment (RazrFalcon) [inline crates/usvg/src/parser/converter.rs:15]: Keep at the old position.

Are you manually moving import or is it IDE/formatter?

File: crates/usvg/src/parser/converter.rs [hunk 3]
```diff
@@ -389,6 +404,29 @@ pub(crate) fn convert_doc(svg_doc: &svgtree::Document, opt: &Options) -> Result<
     Ok(tree)
 }
 
+fn background_path(background_color: svgtypes::Color, area: Rect) -> Option<Path> {
+    let path = {
+        let mut builder = PathBuilder::new();
```
Comment (RazrFalcon) [inline crates/usvg/src/parser/converter.rs:409]: We have `PathBuilder::from_rect`

File: crates/usvg/src/parser/svgtree/names.rs [hunk 4]
```diff
@@ -665,15 +661,16 @@ impl std::fmt::Display for AId {
 struct Map<V: 'static> {
     pub key: u64,
     pub disps: &'static [(u32, u32)],
-    pub entries: &'static [(&'static str, V)],
+    pub entries: &'static[(&'static str, V)],
```
Comment (RazrFalcon) [inline crates/usvg/src/parser/svgtree/names.rs:664]: We should update `crates/usvg/codegen/main.rs` as well then.

File: crates/usvg/src/parser/converter.rs (hunk 2 above)
Comment (LaurenzV) [reply crates/usvg/src/parser/converter.rs:15]: Must be the formatter. :(

File: crates/resvg/tests/integration/main.rs (hunk 1 above)
Comment (LaurenzV) [reply crates/resvg/tests/integration/main.rs]: What do you think about what I did now? Then you can simply do `REPLACE="" cargo test` to indiciate that you want to replace reference images. This way, you don't always have to comment/uncomment it.

Or should I still revert completely?

Comment (RazrFalcon) [reply crates/resvg/tests/integration/main.rs]: It's fine.

**Comment by RazrFalcon** [comment]
Thanks!

**Comment by yisibl** [comment]
This is a bit strange, there is also a shorthand `background` property in CSS. Should we support it?

**Pull request merged**